test/scripts/e2e_subs/assets-app-b.sh

```diff
@@ -40,7 +40,7 @@ XFER4=11
 APP_CREATED_STR='Created app with app index'
 ERR_APP_CL_STR='only clearing out is supported for applications that do not exist'
 ERR_APP_NE_STR='application does not exist'
-ERR_APP_OI_STR1='has not opted in to application'
+ERR_APP_OI_STR1='HTTP 404 Not Found: {"message":"account application info not found"}'
```

Comment:
this error should not get modified : the goal need to be smart enough to identify this use case and emit a backward-compatible message.

Reply:
The Rest API isn't smart enough to identify this use case. It only tells you whether the account resource exists or not and does not differentiate between the app not existing or the user not opting in. Is it worth adding an extra api call to tell whether the app exists or the user hasn't opted in?

Reply:
no - this isn't a REST API issue. The error return by libgoal should contain the inner error message rather than the HTTP error. ( i.e. strip the outer HTTP error )
Separately, I think that "has not opted in to application" is definitely more informative than "account application info not found", since it provides a piece of information that the latter does not contain.

Reply:
"has not opted in to application" is not an accurate error message anymore because it might not be the case